utils/misc/misc.go

```diff
@@ -1289,7 +1268,7 @@ func Unique(stringSlice []string) []string {
 }
 
 func UseFairPickup() bool {
-	return config.GetBool("JobsDB.fairPickup", false) || config.GetBool("EnableMultitenancy", false)
+	return config.GetBool("JobsDB.fairPickup", false) && config.GetString("Router.isolationMode", "default") == "none"
```

Comment:
[note] fair pickup will be used only if it is enabled and isolation mode is `none`